# SQL Part 1 Studio

### Let's practice your SQL quering skills!  For each question, work along in the notebook adding your query and answering the questions. 

### **DataSet**

### <span style="font-size: 14px; font-family: -apple-system, BlinkMacSystemFont, sans-serif;">We will be using the Goodbooks-10k dataset from the exercises in the prep work.&nbsp; Feel free reference your notebook.</span>

- The dataset can be found here: [goodbooks-10k](https://www.kaggle.com/zygmunt/goodbooks-10k)
- You can access the **BooksDB** in the Launchcode server.

### **Business Issue**

You are an employee at a small independet bookstore and you have been tasked to increase sales.  You decide to use the BooksDB to find bookss and themes to highlight in fun promotions throughout each month.  We will be starting with March and then you will have a chance to explore another month of your choosing.  We want to come up with a list of at least 3 events/ promotions to run each month.  If you are looking for ideas to include in your promotions:

- [https://www.calendarr.com/united-states/calendar-2022/](https://www.calendarr.com/united-states/calendar-2022/)
- [https://www.holidayinsights.com/moreholidays/](https://www.holidayinsights.com/moreholidays/)
    - Click on a month and it will take you to a more detailed page

## A. March - Womans History Month, Dr Seuss Day(3/2), National Pie Day3/14, St Patricks Day (3/17), James Patterson(3/22)

Event 1: Womans History Month

Goal: Highlight Popular Womans Authors and Issues

In [7]:
-- SOLUTION
-- Query to look at tags to see which ones may be helpful
SELECT TOP 1000 * 
   FROM BooksDB.dbo.tags
   WHERE tag_name LIKE '%woman' OR tag_name LIKE 'woman%' OR tag_name LIKE '%woman%'
    

(24 rows affected)

Total execution time: 00:00:00.365

tag_id tag_name 937 21-books-every-woman-should-read 1386 a-woman-after-god-s-own-heart 1387 a-woman-after-gods-own-heart 4006 batwoman 4455 biblical-womanhood 4729 black-ops-woman-team-cia-sean-case 5132 books-every-woman-should-read 7194 christian-woman 10986 every-woman-should-read 19038 man-woman 19757 memorable-woman-characters 22269 older-man-younger-woman 22270 older-woman 22271 older-woman-younger-man 23671 pioneer-woman 23891 policewoman-potential-victim 32679 woman-author 32680 woman-s-fiction 32681 woman-s-murder-club 32682 woman-trans-agender-authors 32683 womanhood 32684 womans-literature 32685 womans-murder-club 32740 wonder-woman

Decide I want to explore the following tags:

line 17 tag\_id is woman-author, 32679

line 20 tag\_id is woman-trans-agender-author, 32682

line 21 tag\_id is womanhood, 32683

Decide I want to return the top 10 rated books with each tag to highlight

In [ ]:
SELECT TOP 10(title)
FROM BooksDB.dbo.books
WHERE 

QUESTION 3: Count the number of books where the original\_publication\_year is earlier than 1800.

In [2]:
-- SOLUTION
SELECT COUNT(*)
FROM BooksDB.dbo.books
WHERE original_publication_year < 1800

(1 row affected)

Total execution time: 00:00:00.061

(No column name) 125

QUESTION 4: Create a query that displays distinct authors from the table.

In [3]:
-- SOLUTION
SELECT DISTINCT (authors)
FROM BooksDB.dbo.books


(4664 rows affected)

Total execution time: 00:00:00.410

authors A. Manette Ansay A. Meredith Walters A.A. Milne, Ernest H. Shepard A.C. Gaughen A.G. Howard A.G. Riddle A.J. Banner A.J. Jacobs A.L. Singer, Walt Disney Company A.M. Homes A.N. Roquelaure, Anne Rice A.P.J. Abdul Kalam, Arun Tiwari A.S. Byatt A.S. King A.S.A. Harrison A.W. Tozer Abbi Glines Abigail Haas, Abby McDonald Abraham Verghese Adam Gidwitz Adam Gopnik Adam Hochschild Adam Johnson Adam M. Grant Adam Makos, Larry Alexander Adam Mansbach, Ricardo Cortés Adam Smith Addison Moore Adele Faber, Elaine Mazlish, Kimberly Ann Coe Adeline Yen Mah Adelle Waldman Adolf Hitler Adolfo Bioy Casares, Suzanne Jill Levine, Jorge Luis Borges, Ruth L.C. Simms, Norah Borges de Torre Adrian Conan Doyle, John Dickson Carr Adrian Nicole LeBlanc Adriana Trigiani Aeschylus, Judith Affleck, Philip de May, Patricia E. Easterling, John Harrison Aeschylus, Robert Fagles, William Bedell Stanford Aesop, Laura Harris, Laura Gibbs Agatha Christie Agatha Christie, Hugh Fraser Agatha Christie, Robin Bailey Agatha Christie, Ροζίτα Σώκου Ahmad Fuadi Ai Yazawa Aimee Bender Aimee Carter Akira Amano, JN Productions, Frances E. Wall Akira Toriyama Al Dempster, Milt Banta, Golden Books, Walt Disney Company Al Franken Al Perkins, Eric Gurney Al Ries, Jack Trout Alaa Al Aswany, علاء الأسواني Alain de Botton Alan Richardson, Karen Tack Alan Bennett Alan Bradley Alan Brennert Alan Cumming Alan Dean Foster Alan Dean Foster, George Lucas Alan Dean Foster, Lawrence Kasdan, J.J. Abrams, Michael Arndt Alan Hollinghurst Alan Lee, Ian McKellen Alan Lightman Alan Moore, Brian Bolland, Tim Sale Alan Moore, Curt Swan, George Pérez, Kurt Schaffenberger Alan Moore, Dave Gibbons, John Higgins Alan Moore, David Lloyd Alan Moore, Eddie Campbell, Pete Mullins Alan Moore, Kevin O'Neill Alan Moore, Stephen R. Bissette, John Totleben, Rick Veitch, Ramsey Campbell Alan Paton Alan W. Watts Alan Weisman Alastair Reynolds Albert Camus, Justin O'Brien Albert Camus, Matthew Ward Albert Camus, Stuart Gilbert Albert Einstein, Nigel Calder, Robert W. Lawson Aldo Leopold Aldous Huxley Aldous Huxley, Christopher Hitchens Aleatha Romig Aleksandr Griboyedov, Александр Сергеевич Грибоедов Aleksandr Solzhenitsyn, Edward E. Ericson Jr. Aleksandr Solzhenitsyn, H.T. Willetts Alessandra Torre Alessandro Baricco Alex Sanchez Alex Ross Alex Berenson Alex Flinn Alex Garland Alex Gino Alex Haley Alex Kava Alex Kotlowitz Alex Lake Alex Marwood Alex Rosenberg Alexander Gordon Smith Alexander Hamilton, James Madison, John Jay, Philo-Publius, Clinton Rossiter, Charles R. Kessler Alexander McCall Smith Alexander Osterwalder, Yves Pigneur Alexander Pushkin, James E. Falen Alexandra Adornetto Alexandra Bracken Alexandra Fuller Alexandra Horowitz Alexandra Potter Alexandra Ripley Alexandre Dumas Alexandre Dumas fils, David Coward Alexandre Dumas, David Coward, Auguste Maquet Alexandre Dumas, Francine du Plessix Gray, Joachim Neugroschel Alexandre Dumas, Robin Buss Alexis de Tocqueville, Isaac Kramnick, Gerald Bevan Alfred Bester Alfred Bester, Neil Gaiman Alfred Lansing Alfred Tennyson, Charles Keeping Alfred Uhry Ali Novak, Fallzswimmer Ali Smith Alice Miller, Ruth Ward Alice Clayton Alice Hoffman Alice LaPlante Alice McDermott Alice Munro Alice Schroeder Alice Sebold Alice Walker Alice Waters Alisa Kwitney, Neil Gaiman Alisa Valdes, Alisa Valdes-Rodriguez Alison Bechdel Alison Croggon Alison Goodman Alison Weir Alissa Nutting Alistair MacLean Alistair MacLean, Clint Willis Allan Pease, Barbara Pease Allen Eskens Allen Ginsberg, William Carlos Williams Allen Say Allie Brosh Allison Pearson Ally Carter Ally Condie Alton Brown Alvin Schwartz, Stephen Gammell Alwyn Hamilton Alyson Noel Alyson Richman Amanda Berry, Gina DeJesus, Mary Jordan, Kevin Sullivan Amanda Brown Amanda Coplin Amanda Foreman Amanda Grange Amanda Hocking Amanda Knox Amanda Lindhout, Sara Corbett Amanda Palmer Amazon Ambrose Bierce Amelia Atwater-Rhodes Amélie Nothomb American Psychiatric Association America's Test Kitchen, Carl Tremblay, D

QUESTION 5: Create a query that displays a count of all the books that contain a language code for English.  This could be represented in the table as "eng" or "en-".

In [55]:
-- Solution

-- SELECT COUNT(*)
   -- FROM BooksDB.dbo.books
   -- WHERE language_code LIKE 'en%';

-- rows: 8730

Commands completed successfully.

Total execution time: 00:00:00.052

QUESTION 6: Create a query to check how many original titles were written during World War I era (1914-1921).

In [58]:
-- Solution
-- --SELECT original_title
    -- FROM BooksDB.dbo.books
    -- WHERE original_publication_year BETWEEN 1914 AND 1921;
    -- WHERE original_publication_year IN(1914,1915,1916,1917,1918,1919,1920,1921);

-- 39 rows

Commands completed successfully.

Total execution time: 00:00:00.048

# Part 2 Choose Another Month

Choose another month and plan at least 3 events / promotions and answer the following questions.

Which month did you choose?  

What 3 events / promotions are you highlighting?

> 1.

> 2.

> 3.

# Part 3. Summarize your Work

For each event write at least one query to support your choice and record you thoughts as to why you used the paticlular query.

In [ ]:
-- Event 1

In [ ]:
-- Event 2

In [ ]:
-- Event 3